R/germline.R

```diff
@@ -122,10 +131,16 @@ generate_germline_sequence <- function(seq, v_ref, v_end, cdr3_start, cdr3_end,
 
     cdr3_part <- paste(rep("n", cdr3_end - cdr3_start), collapse = "")
 
-    j_part <- stringr::str_sub(seq, max(cdr3_end, j_start))
+    if (j_start >= cdr3_end) {
+      j_part <- j_ref
+    } else {
+      # trim intersection of J and CDR3 from reference J gene
+      j_part <- stringr::str_sub(j_ref, cdr3_end - j_start + 1)
```

Comment:
it is not obligatory here to use package::function construction, but it is normal because of improving of readability of code